LibreNMS/Data/Store/Rrd.php

```diff
@@ -442,6 +444,67 @@ public function buildCommand($command, $filename, $options)
         return "$command $filename $options";
     }
 
+    /**
+     * Get array of all rrd files for a device,
+     * via rrdached or localdisk.
+     *
+     * @param array $device device for which we get the rrd's
+     * @return array $rrdfilearray array of rrd files for this host
+     */
+    public function getRrdFiles($device)
+    {
+        if ($this->rrdcached) {
+            $filename = sprintf('/%s', $device['hostname']);
+            $rrd_files = $this->command('list', $filename, '');
+            // Command output is an array, create new array with each filename as a item in array.
+            $rrdfilearray = preg_split('/\s+/', trim($rrd_files[0]));
+        } else {
+            $rrddir = self::dirFromHost($device['hostname']);
```

Comment:
Should be `$rrddir = $this->dirFromHost($device['hostname']);`

Rrd is a singleton that stays loaded in memory after it is started.   The external alias \Rrd is a static way to access that singleton.